dwave/gate/simulator/operation_generation.py

```diff
@@ -466,6 +534,146 @@ def generate_op(
     )
 
 
+def generate_op_set(
+    op_name: str,
+    num_targets: int,
+    num_controls: int,
+    operators: List[np.ndarray],
+) -> OpInfo:
+    """Generate C code and cython wrapper for a fast implementation of the given
+    set of operators. These should follow the rule (where {E_k} is the set of operators)
+
+        \sum_k E_k^{\dagger} E_k = I.
+
+    The operators are applied one by one, and the norm of the resulting state is
+    computed. Then, one of the operators is chosen probabilistically according to the
+    relative probabilities as determined by the norms. Finally, this operator is applied
+    to the state (if `apply_operator` is set to True). The index of the chosen operator
+    is returned.
+
+    This should be useful both for generating measurement operations as well as
+    simulating noise with Kraus operators.
+
+    See :func:`generate_op_c_code` for more detail of how each operator
+    is generated/compiled.
+
+    This will generate both little and big-endian versions of the operator sets.
+
+    Args:
+        op_name:
+            A name for the gate which will be used in the names of the C functions.
+
+        num_targets: Number of targets (non-control qubits) of the gate.
+
+        num_controls: Number of controls for the operator set.
+
+        little_endian:
+            If true, use little-endian convention for the basis states. Otherwise, use
+            big-endian.
+
+        operators:
+            The list of operators in the form of numpy arrays.
+
+    Returns:
+        C code/cython wrapper that implement the gate.
+
+    """
+
+    op_sum = sum(np.conj(op).T.dot(op) for op in operators)
+    if not np.all(np.isclose(op_sum, np.eye(2**num_targets))):
+        raise ValueError("sum of operators by their conjugates must sum to identity")
+
+    c_codes = []
+
+    import itertools
+
+    full_op_options = {
+        (op_idx, little_endian, collect_norm):
+            f"{op_name}_op{op_idx}_{['norm', 'apply'][collect_norm]}_{['big', 'little'][little_endian]}_endian"
```

Comment:
I think `'apply'` and `'norm'` are in the wrong order here, no?
```suggestion
            f"{op_name}_op{op_idx}_{['apply, 'norm'][collect_norm]}_{['big', 'little'][little_endian]}_endian"
```

Reply:
Yes! Fixed